### PR TITLE
[Behat][Product] Product deletion scenarios

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | yes
| Deprecations?   | no
| Related tickets | partially #4140 
| License         | MIT

### CODE REVIEW

File: features/product/deleting_product.feature [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    @todo
+    Scenario: Deleted product variants disappear from the registry
+        When I delete the "Toyota GT86 model" product
+        Then there should be no "1:43" variant in the registry
+
+    @todo
+    Scenario: Deleted product reviews disappear form the registry
+        When I delete the "Toyota GT86 model" product
+        Then there should be no reviews of this product
+
```
Comment (Zales0123) [inline features/product/deleting_product.feature]: Redundant blank line ;)

File: features/product/deleting_product.feature [hunk 2]
```diff
@@ -0,0 +1,26 @@
+@product
+Feature: Deleting a product
+    In order to remove test or incorrect products
```
Comment (michalmarcinkowski) [inline features/product/deleting_product.feature]: We agreed on one version:
`In order to remove test, obsolete or incorrect {resource}`

File: features/product/deleting_product.feature [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    As an Administrator
+    I want to be able to delete products from the registry
```
Comment (michalmarcinkowski) [inline features/product/deleting_product.feature]: product from the **product catalog**

File: features/product/deleting_product.feature [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        When I delete the "Toyota GT86 model" product
+        Then this product should not exist in the product catalog
+
+    @todo
+    Scenario: Deleted product variants disappear from the product catalog
+        When I delete the "Toyota GT86 model" product
+        Then there should be no "1:43" variant in the variant catalog
+
+    @todo
+    Scenario: Deleted product reviews disappear form the product catalog
```
Comment (pjedrzejewski) [inline features/product/deleting_product.feature]: from

File: features/product/deleting_product.feature [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+    @todo
+    Scenario: Deleted product disappears from the product catalog
+        When I delete the "Toyota GT86 model" product
+        Then this product should not exist in the product catalog
+
+    @todo
+    Scenario: Deleted product variants disappear from the product catalog
+        When I delete the "Toyota GT86 model" product
+        Then there should be no "1:43" variant in the variant catalog
```
Comment (lchrusciel) [inline features/product/deleting_product.feature]: `variant catalog` => `product catalog`

File: features/product/deleting_product.feature [hunk 6]
```diff
@@ -0,0 +1,26 @@
+@product
+Feature: Deleting a product
+    In order to remove test, obsolete or incorrect products
+    As an Administrator
+    I want to be able to delete products from the product catalog
+
+    Background:
+        Given the store operates on a single channel in "France"
+        And the store has a product "Toyota GT86 model"
+        And the product "Toyota GT86 model" has "1:43" variant priced at €15.00
```
Comment (lchrusciel) [inline features/product/deleting_product.feature]: :+1:

File: features/product/deleting_product.feature (hunk 5 above)
Comment (michalmarcinkowski) [inline features/product/deleting_product.feature]: ... variant **of this product** in the ...

**Comment by michalmarcinkowski** [comment]
:+1:

**Pull request merged**